File: repos_estudiantes_TARQUI_primera_entrega/LUNA_CRUZ_CAMILA_MILENKA/POLIMORFISMO/Ejercicio5/Universidad.py
#5.Se hace referencia a algunos de los diferentes ambientes 
#de la Universidad mediante las siguientes clases:
class Oficina:
    def __init__(self, sillas, escritorios, estanterias):
        self.nroSillas = sillas
        self.nroEscritorios = escritorios
        self.nroEstanterias = estanterias

    # Método para mostrar datos de la oficina
    def mostrar(self):
        print("Sillas:", self.nroSillas)
        print("Escritorios:", self.nroEscritorios)
        print("Estanterías:", self.nroEstanterias)

    # Sobrecarga del método cantidadMuebles
    def cantidad_muebles(self, sillas=None, escritorios=None, estanterias=None):
        if sillas is not None and escritorios is not None and estanterias is not None:
            return sillas + escritorios + estanterias
        return self.nroSillas + self.nroEscritorios + self.nroEstanterias


class Aula:
    def __init__(self, nombre, capacidad, pupitres):
        self.nombre = nombre
        self.capacidad = capacidad
        self.nroPupitres = pupitres

    # Método para mostrar datos del aula
    def mostrar(self):
        print("Aula:", self.nombre)
        print("Capacidad:", self.capacidad)
        print("Pupitres:", self.nroPupitres)

    # Sobrecarga del método cantidadMuebles
    def cantidad_muebles(self, pupitres=None):
        if pupitres is not None:
            return pupitres
        return self.nroPupitres


class Laboratorio:
    def __init__(self, nombre, capacidad, mesas, sillas):
        self.nombre = nombre
        self.capacidad = capacidad
        self.nroMesas = mesas
        self.nroSillas = sillas

    # Método para mostrar datos del laboratorio
    def mostrar(self):
        print("Laboratorio:", self.nombre)
        print("Capacidad:", self.capacidad)
        print("Mesas:", self.nroMesas)
        print("Sillas:", self.nroSillas)

    # Sobrecarga del método cantidadMuebles
    def cantidad_muebles(self, mesas=None, sillas=None):
        if mesas is not None and sillas is not None:
            return mesas + sillas
        return self.nroMesas + self.nroSillas


if __name__ == "__main__":
    # Instanciar 2 objetos Oficina, 2 Aulas y 1 Laboratorio
    ofi1 = Oficina(8, 4, 4)
    ofi2 = Oficina(10, 5, 3)

    aula1 = Aula("401", 20, 20)
    aula2 = Aula("502", 20, 20)

    lab = Laboratorio("ESTADISTICA", 15, 10, 16)

    print("-----OFICINAS------")
    print("La Oficina tiene:", ofi1.cantidad_muebles(6, 8, 3))
    print("--------------------------------")
    ofi2.mostrar()
    print("Total de los muebles:", ofi2.cantidad_muebles())
    print("--------------------------------")
    print("------AULAS------")
    aula1.mostrar()

File: repos_estudiantes_TARQUI_primera_entrega/LUNA_CRUZ_CAMILA_MILENKA/POLIMORFISMO/Ejercicio5/Universidad.java
package POLIMORFISMO.Ejercicio5;
//5.Se hace referencia a algunos de los diferentes ambientes 
//de la Universidad mediante las siguientes clases:

class Oficina {
    private int nroSillas, nroEscritorios, nroEstanterias;

    public Oficina(int sillas, int escritorios, int estanterias) {
        this.nroSillas = sillas;
        this.nroEscritorios = escritorios;
        this.nroEstanterias = estanterias;
    }
    //b)Crear un método mostrar() para mostrar los datos de cada objeto
    public void mostrar() {
        System.out.println("Sillas: " + nroSillas);
        System.out.println("Escritorios: " + nroEscritorios);
        System.out.println("Estanterias: " + nroEstanterias);
    }
    //c)Sobrecargar el método cantidadMuebles(), para conocer el número total de muebles dentro de cada ambiente
    public int cantidadMuebles(int sillas, int escritorios, int estanterias) {
        return sillas + escritorios + estanterias;
    }

    public int cantidadMuebles() {
        return cantidadMuebles(nroSillas, nroEscritorios, nroEstanterias);
    }
}
class Aula {
    private String nombre;
    private int capacidad, nroPupitres;

    public Aula(String nombre, int capacidad, int pupitres) {
        this.nombre = nombre;
        this.capacidad = capacidad;
        this.nroPupitres = pupitres;
    }
    //b)Crear un método mostrar() para mostrar los datos de cada objeto
    public void mostrar() {
        System.out.println("Aula: " + nombre);
        System.out.println("Capacidad: " + capacidad);
        System.out.println("Pupitres: " + nroPupitres);
    }
    //c)Sobrecargar el método cantidadMuebles(), para conocer el número total de muebles dentro de cada ambiente
    public int cantidadMuebles(int pupitres) {
        return pupitres;
    }

    public int cantidadMuebles() {
        return cantidadMuebles(nroPupitres);
    }
}
class Laboratorio {
    private String nombre;
    private int capacidad, nroMesas, nroSillas;

    public Laboratorio(String nombre, int capacidad, int mesas, int sillas) {
        this.nombre = nombre;
        this.capacidad = capacidad;
        this.nroMesas = mesas;
        this.nroSillas = sillas;
    }
    //b)Crear un método mostrar() para mostrar los datos de cada objeto
    public void mostrar() {
        System.out.println("Laboratorio: " + nombre);
        System.out.println("Capacidad: " + capacidad);
        System.out.println("Mesas: " + nroMesas);
        System.out.println("Sillas: " + nroSillas);
    }
    //c)Sobrecargar el método cantidadMuebles(), para conocer el número total de muebles dentro de cada ambiente
    public int cantidadMuebles(int mesas, int sillas) {
        return mesas + sillas;
    }
    
    public int cantidadMuebles() {
        return cantidadMuebles(nroMesas, nroSillas);
    }
    
    public static void main(String[] args) {

        //a)Instanciar 2 objetos Oficina, 2 Aulas y 1 Laboratorio
        
        Oficina ofi1 = new Oficina(8, 4, 4);
        Oficina ofi2 = new Oficina(10, 5, 3);

        Aula aula1 = new Aula("401", 20, 20);
        Aula aula2 = new Aula("502", 20, 20);

        Laboratorio lab = new Laboratorio("ESTADISTICA", 15, 10, 16);

        System.out.println("-----OFICINAS------");
        System.out.println("La Oficina tiene: " + ofi1.cantidadMuebles(6, 8, 3));
        System.out.println("--------------------------------");
        ofi2.mostrar();
        System.out.println("Total de los muebles: " + ofi2.cantidadMuebles());
        System.out.println("--------------------------------");
        System.out.println("------AULAS------");
        aula1.mostrar();
        System.out.println("Total de los muebles: " + aula1.cantidadMuebles());
        System.out.println("--------------------------------");
        aula2.mostrar();
        System.out.println("Total de los muebles: " + aula2.cantidadMuebles());
        System.out.println("--------------------------------");
        System.out.println("-----LABORATORIO------");
        lab.mostrar();
        System.out.println("Total de los muebles: " + lab.cantidadMuebles());
    }
}

